clang/lib/AST/ByteCode/Compiler.cpp

```diff
@@ -1267,7 +1268,15 @@ bool Compiler<Emitter>::VisitVectorBinOp(const BinaryOperator *E) {
   auto getElem = [=](unsigned Offset, unsigned Index) {
     if (!this->emitGetLocal(PT_Ptr, Offset, E))
       return false;
-    return this->emitArrayElemPop(ElemT, Index, E);
+    if (!this->emitArrayElemPop(ElemT, Index, E))
+      return false;
+    if (E->isLogicalOp()) {
+      if (!this->emitPrimCast(ElemT, PT_Bool, Ctx.getASTContext().BoolTy, E))
+        return false;
+      if (!this->emitPrimCast(PT_Bool, ResultElemT, VecTy->getElementType(), E))
+        return false;
+    }
```

Comment:
I hate the double cast, but I have no better idea either :/

Reply:
Yes, this approach doesn't looks well. Seems BitAnd/BitOr doesn't support bool type, should we extend these two operations to support bool type? Then we can avoid these casts.

Reply:
I think that would just move the cast further down, i.e. you still have to convert from bool to` ResultT`. I guess this is fine for now.